File: app/src/main/java/com/lutech/flashlight/Notification/NotificationListener.java
package com.lutech.flashlight.Notification;

import android.app.Notification;
import android.content.Context;
import android.os.Bundle;
import android.service.notification.NotificationListenerService;
import android.service.notification.StatusBarNotification;
import android.util.Log;

import androidx.core.app.NotificationCompat;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class NotificationListener extends NotificationListenerService {
    private static final String TAG = "NotificationListener";
    private static final String FB_PACKAGE = "com.facebook.orca";
    private static final String WA_PACKAGE = "com.whatsapp";
    private static final String ZA_PACKAGE = "com.zing.zalo";
    private static final String WE_PACKAGE = "com.whatsapp";
    private static final String VIBER_PACKAGE = "com.whatsapp";
    private static final String TE_PACKAGE = "com.whatsapp";
    private static final String SKY_PACKAGE = "com.whatsapp";
    Context context;


    @Override
    public void onListenerConnected() {
        Log.i(TAG, "Notification Listener connected");
    }


    @Override
    public void onCreate() {
        super.onCreate();
        context = getApplicationContext();
    }

    @Override
    public void onNotificationPosted(StatusBarNotification sbn) {
        if (!sbn.getPackageName().equals(FB_PACKAGE) && !sbn.getPackageName().equals(WA_PACKAGE) && !sbn.getPackageName().equals(ZA_PACKAGE))
            return;
        Notification notification = sbn.getNotification();
        Bundle bundle = notification.extras;

        String sender = bundle.getString(NotificationCompat.EXTRA_TITLE);
        String message = bundle.getString(NotificationCompat.EXTRA_TEXT);
        String app;

//            app = getString(R.string.txt_facebook);
    }


}

File: app/src/main/java/com/lutech/flashlight/Notification/NotificationListener.kt
package com.lutech.flashlight.Notification

import android.content.Context
import android.content.Intent
import android.os.Build
import android.os.Handler
import android.provider.Settings
import android.provider.Telephony
import android.service.notification.NotificationListenerService
import android.service.notification.StatusBarNotification
import android.util.Log
import androidx.annotation.NonNull
import androidx.annotation.Nullable
import com.lutech.flashlight.ads.Constants
import com.lutech.flashlight.camera.CameraTorchListener
import com.lutech.flashlight.camera.MyCameraImpl
import com.lutech.flashlight.camera.MyCameraImpl.Companion.newInstance
import com.lutech.flashlight.util.MySharePreference
import org.greenrobot.eventbus.EventBus


class NotificationListener : NotificationListenerService() {
    private var mySharePreference: MySharePreference? = null
    private var mCameraImpl: MyCameraImpl? = null
    private var eventBus: EventBus? = null

    private var mIsFlashlightOn = false
    var mContext: Context? = null
    override fun onListenerConnected() {
        Log.i(TAG, "Notification Listener connected")
    }

    override fun onCreate() {
        super.onCreate()
        mContext = this
        mySharePreference = MySharePreference(this)
    }

    override fun onNotificationPosted(sbn: StatusBarNotification) {
        val isAlertSms = mySharePreference!!.getFlashAlert(Constants.ALERT_SMS)!!.isStatusChecked
        val isAlertNotifi =
            mySharePreference!!.getFlashAlert(Constants.ALERT_NOTIFICATION)!!.isStatusChecked
        var mIsFlashlightOn: Boolean
        eventBus = EventBus.getDefault()
        if (!isAlertSms && !isAlertNotifi) {
            return
        }
        Log.d(TAG, "onNotificationPosted: " + sbn.packageName)

        if (sbn.packageName == "sms_default_application" || sbn.packageName == "com.google.android.apps.messaging") {
            if (isAlertSms) {
                Log.d(TAG, "onNotificationPosted:setupSMS")
                setupCameraImpl(Constants.ALERT_SMS)
            }
        } else {
            Log.d(TAG, "onNotificationPosted: " + isAlertNotifi)

            if (isAlertNotifi) {
                Log.d(TAG, "onNotificationPosted:setUpNotifi")

                setupCameraImpl(Constants.ALERT_NOTIFICATION)
            }
        }

    }

    private fun setupCameraImpl(type: String) {
        mCameraImpl = newInstance(this, object : CameraTorchListener {
            override fun onTorchEnabled(isEnabled: Boolean) {

            }

            override fun onTorchUnavailable() {
                mCameraImpl!!.onCameraNotAvailable()
                Log.d(TAG, "onTorchUnavailable: ")
            }
        }, type,false)
        mCameraImpl!!.handleCameraSetup()
        mIsFlashlightOn = mCameraImpl!!.toggleStroboscope()
        Handler().postDelayed({
            mCameraImpl?.stopStroboscope()
        }, 5000)

    }

    @Nullable
    fun getDefaultSmsAppPackageName(@NonNull context: Context): String? {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) try {
            return Telephony.Sms.getDefaultSmsPackage(context)
        } catch (e: Throwable) {
        }
        val intent = Intent(Intent.ACTION_VIEW)
            .addCategory(Intent.CATEGORY_DEFAULT).setType("vnd.android-dir/mms-sms")
        val resolveInfoList = context.packageManager.queryIntentActivities(intent, 0)
        return if (!resolveInfoList.isEmpty()) resolveInfoList[0].activityInfo.packageName else null
    }

    override fun onDestroy() {
        super.onDestroy()
        releaseCamera()
    }

    private fun releaseCamera() {
        if (mCameraImpl != null) {
            mCameraImpl!!.releaseCamera()
            mCameraImpl = null
        }
    }

    companion object {
        private const val TAG = "NotificationListener"

    }

}